features/netsocket/Socket.cpp

```diff
@@ -83,11 +94,39 @@ int Socket::modify_multicast_group(const SocketAddress &address, nsapi_socket_op
 
 int Socket::join_multicast_group(const SocketAddress &address)
 {
+    if(_multicast_addr_count == SOCKET_MAX_MULTICAST_MEMBERSHIPS) {
```

Comment:
This is the wrong layer - it's in the "friendly veneer" rather than the core socket option. It means direct use of the socket options won't track.